### PR TITLE
sanitize tar archives due to CVE

### PR DESCRIPTION
**Pull request by osherdp**
Patching CVE-2007-4559
(https://github.com/advisories/GHSA-gw9q-c7gh-j9vm) which might allow attackers to overwrite arbitrary files which are not under destination path.

Implementation inspiration: https://github.com/openshift/assisted-test-infra/pull/1898

### CODE REVIEW

File: src/assisted_test_infra/test_infra/utils/logs_utils.py [hunk 1]
```diff
@@ -12,6 +15,25 @@
 OC_DOWNLOAD_LOGS_TIMEOUT = 60 * 60
 
 
+@contextlib.contextmanager
+def _safe_open_tar(tar_path: Union[str, Path], destination_path: Union[str, Path]) -> None:
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/utils/logs_utils.py:19]: You can now do:
```suggestion
def _safe_open_tar(tar_path: str | Path, destination_path: str | Path) -> None:
```

File: src/assisted_test_infra/test_infra/utils/logs_utils.py [hunk 2]
```diff
@@ -140,39 +161,36 @@ def _verify_oc_logs_uploaded(cluster_tar_path):
 
 
 def _verify_node_logs_uploaded(dir_path, file_path):
-    gz = tarfile.open(os.path.join(dir_path, file_path))
-    logs = gz.getnames()
-    for logs_type in ["agent.logs", "installer.logs", "mount.logs"]:
-        assert any(logs_type in s for s in logs), f"{logs_type} isn't found in {logs}"
-    gz.close()
+    with tarfile.open(os.path.join(dir_path, file_path)) as gz:
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/utils/logs_utils.py:164]: Why not use `_safe_open_tar`?

Comment (osherdp) [reply src/assisted_test_infra/test_infra/utils/logs_utils.py:164]: because it doesn't do extraction, only reads the names and asserts their values

**Review by eliorerz — APPROVED** [review]
LGTM

**Pull request merged**